Questions

TheJarles opened the pull request:
Same as the earlier pull request ([https://github.com/kolibre/kobra/pull/4](url)) but amended with a bug fix and removed a merge conflict I had forgotten.

jabbors commented on app/models/user_content.rb:
```diff
@@ -2,6 +2,10 @@ class UserContent < ApplicationRecord
   belongs_to :user
   belongs_to :content
   belongs_to :content_list
+<<<<<<< HEAD
```
This change shall be reverted of fixed. Originates from an unresolved merge conflict.

jabbors commented on db/migrate/20190221181952_create_question_texts.rb:
```diff
@@ -0,0 +1,10 @@
+  def change
+    create_table :question_texts do |t|
+      t.references :language, foreign_key: true
+      t.integer :text, null: false
```
Bad type in the reference database, shall be `text`.

jabbors commented on db/schema.rb:
```diff
@@ -72,6 +100,59 @@
+  end
+
+  create_table "question_types", force: :cascade do |t|
+    t.string "type_name", null: false
```
Let's simplify the column name to `name`. It's implied from the table name that it's a type of some kind.

TheJarles commented:
I implemented the requested changes. I also took the liberty to change some of the types of the columns in QuestionInput (text_numeric and text_alphanumeric => text instead of integer, and default value for allow_multiple_selections from nil to 0 as it is a integer-represented boolean, correct?). I also noticed that the whitelisted variables names in the  models' respective controllers were not updated after initially generated, so I updated those as well.

jabbors commented on db/migrate/20190221181516_create_question_inputs.rb:
```diff
@@ -0,0 +1,14 @@
+    create_table :question_inputs do |t|
+      t.references :question, foreign_key: true
+      t.integer :allow_multiple_selections, default: 0
+      t.text :text_numeric, default: nil
```
`text_numeric` and `text_alphanumeric` are not used to store input from the user, the purpose is instead to define which input types are supported for a question sent from the server to the user.

I checked the code and we only check if the value is null or not when we build the input questions. Thus it does not really matter of which type they are but to be consistent and save a few bytes we shall change them back to integers.

TheJarles replied:
I reverted the type changes, so now I just need a new approval for the pull request.

jabbors replied:
You did not actually revert anything, instead you change another column from integer to text 😄 

So now you have two reverts to do 😄 In this table, fields `text_numeric`, `text_alphanumeric` and `audio` should be of type integer. Not that it really matters be we save a few bytes and that's an optimisation that is better to do know than later.

TheJarles replied:
Hahah, that's what happens when you're unfocused. I'll fix it in a bit.
